### ISSUE TITLE
Cannot Pair Mac from Visual Studio

### ISSUE THREAD

**Issue by creeksmarts-jon** [xamarin/xamarin-macios#3688]
### Steps to Reproduce

1. Open any iOS solution in Visual Studio, such as Hello.iOS_Multiscreen
2. Click the Connect button to Pair a Mac
3. The error appears: 

> One or more Xcode tools were not installed successfully. Failed tools: XcodeExtensionSupport.pkg, MobileDevice.pkg. Even if you can continue working with iOS apps, we strong recommend you to manually open Xcode on 'MacBook' and install them if asked.

### Expected Behavior
The PC connects to the Mac and builds the solution in iOS simulator.

### Actual Behavior
The Mac and the PC are not connected, and building to the Mac is broken.

When I open Xcode, there are no prompts. I verified my certificates are valid in Xcode. I was able to build yesterday before today's updates. I updated both the Mac and PC.

### Environment

```
Microsoft Visual Studio Community 2017 
Version 15.6.0
VisualStudio.15.Release/15.6.0+27428.1
Microsoft .NET Framework
Version 4.7.02556

Installed Version: Community

Visual F# Tools 10.1 for F# 4.1   00369-60000-00001-AA395
Microsoft Visual F# Tools 10.1 for F# 4.1

Application Insights Tools for Visual Studio Package   8.11.10212.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2017   15.0.40214.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   1.0
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2012   4.0.21208.0
For additional information, visit https://www.asp.net/

ASP.NET Web Frameworks and Tools 2017   5.2.51214.0
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   15.0.40215.0
Azure App Service Tools v3.0.0

C# Tools   2.7.0-beta3-62620-07. Commit Hash: e873e69306527a0424e97af2985370556f474019
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub.VisualStudio   2.4.2.1331
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Merq   1.1.17-rc (cba4571)
Command Bus, Event Stream and Async Manager for Visual Studio extensions.

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2017 - v2.9.51212.2

Microsoft Continuous Delivery Tools for Visual Studio   0.3
Simplifying the configuration of continuous build integration and continuous build delivery from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Mono Debugging for Visual Studio   4.9.11-pre (71eb098)
Support for debugging Mono processes with Visual Studio.

Node.js Tools   1.4.11027.3
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   4.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   15.1.61801.210
Microsoft SQL Server Data Tools

TypeScript Tools   15.6.20202.3
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   2.7.0-beta3-62620-07. Commit Hash: e873e69306527a0424e97af2985370556f474019
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Apache Cordova   15.123.7102.2
Visual Studio Tools for Apache Cordova

Visual Studio Tools for Unity   3.6.0.4
Visual Studio Tools for Unity

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VisualSVN   6.0.4
Integration with Subversion version control. For more information about VisualSVN, see the VisualSVN website at http://www.visualsvn.com
Copyright © 2017 VisualSVN Software Ltd. All rights reserved.

Xamarin   4.9.0.749 (9b0fce36d)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   4.10.58 (cee1369d0)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android SDK   8.2.0.15 (HEAD/22d97e153)
Xamarin.Android Reference Assemblies and MSBuild support.

Xamarin.iOS and Xamarin.Mac SDK   11.8.0.20 (1c6f300)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Build Logs

Xamarin.Messaging.Integration.State.ServerStateContext|Information|0|Xcode 'MobileDeviceDevelopment.pkg' package is already installed
Xamarin.Messaging.Integration.State.ServerStateContext|Information|0|Checking Xcode 'XcodeExtensionSupport.pkg' package installation...
Xamarin.Messaging.Ssh.SshCommandRunner|Warning|0|Failed to execute 'pkgutil --pkg-info com.apple.pkg.XcodeExtensionSupport': ExitStatus = 1
Xamarin.Messaging.Ssh.SshCommandRunner|Warning|0|No receipt for 'com.apple.pkg.XcodeExtensionSupport' found at '/'.
Xamarin.Messaging.Integration.State.ServerStateContext|Information|0|Installing Xcode package 'XcodeExtensionSupport.pkg'...

**Comment by brendanzagaeski** [comment]
Tallying other user reports for bookkeeping: 1 or 2 users in <https://forums.xamarin.com/discussion/123419> (2018-03-06 to 2018-03-07)

Note for @creeksmarts-jon, the reporter of that forum thread mentioned:
>  Got VS Connection dialog to quit complaining by opening up Xcode.app, Navigating to the /Packages directory and manually installing MobileDevice.pkg.

... in case that might help resolve the error about "MobileDevice.pkg" in your case too.

**Comment by creeksmarts-jon** [comment]
Thanks. From the log, it looks like MobileDeviceDevelopment package is already installed but that the second package cannot be installed.

This seems like the most relevant part from the log:

> Failed to execute 'pkgutil --pkg-info com.apple.pkg.XcodeExtensionSupport': ExitStatus = 1

Does that _pkgutil_  mean anything to you?

**Comment by brendanzagaeski** [comment]
**Cross-referencing note to the Xamarin team**

Also reported in Developer Community:

<https://developercommunity.visualstudio.com/content/problem/209704/vs-156-xcode-tools-installation-fails.html>

Unfortunately at the moment that report also lists a second unrelated symptom. That could lead to some confusion during the discussion on that report.

**Comment by mauroa** [comment]
The corresponding feedback item (https://developercommunity.visualstudio.com/content/problem/209704/vs-156-xcode-tools-installation-fails.html) has been closed because for 15.7 the validation of Xcode Tools packages from VS is silent, and if they can't be updated then there is no warning shown to the user, which was confusing.
This has been changed that way because there is no action item from VS if the update cannot be performed and it also shouldn't affect the development process.
That said, closing this issue to match the feedback item state.
